Tests/TransferMethod/HyperwalletTransferMethodTests.swift

```diff
@@ -53,32 +53,32 @@ class HyperwalletTransferMethodTests: XCTestCase {
         XCTAssertNotNil(linkNext?.href)
 
         // check the bank card
-        let bankCard = transferMethods?.data.first { ($0.getField(fieldName: .type) as? String ?? "")  == "BANK_CARD" }
-        XCTAssertEqual(bankCard?.getField(fieldName: .type) as? String, "BANK_CARD")
-        XCTAssertEqual(bankCard?.getField(fieldName: .token) as? String, "trm-00002")
+        let bankCard = transferMethods?.data.first { ($0.type ?? "")  == "BANK_CARD" }
+
+        XCTAssertEqual(bankCard?.type, "BANK_CARD")
+        XCTAssertEqual(bankCard?.token, "trm-00002")
+        XCTAssertEqual(bankCard?.status, "ACTIVATED")
+        XCTAssertEqual(bankCard?.transferMethodCountry, "US")
+        XCTAssertEqual(bankCard?.transferMethodCurrency, "USD")
         XCTAssertEqual(bankCard?.getField(fieldName: .cardNumber) as? String, "************1358")
         XCTAssertEqual(bankCard?.getField(fieldName: .dateOfExpiry) as? String, "2022-12")
         XCTAssertEqual(bankCard?.getField(fieldName: .cardType) as? String, "DEBIT")
         XCTAssertEqual(bankCard?.getField(fieldName: .cardBrand) as? String, "VISA")
-        XCTAssertEqual(bankCard?.getField(fieldName: .status) as? String, "ACTIVATED")
-        XCTAssertEqual(bankCard?.getField(fieldName: .transferMethodCountry) as? String, "US")
-        XCTAssertEqual(bankCard?.getField(fieldName: .transferMethodCurrency) as? String, "USD")
 
         // check the bank account
-        let bankAccount = transferMethods?.data.first {
-            ($0.getField(fieldName: .type) as? String ?? "")  == "BANK_ACCOUNT"
-        }
-        XCTAssertEqual(bankAccount?.getField(fieldName: .type) as? String, "BANK_ACCOUNT")
-        XCTAssertEqual(bankAccount?.getField(fieldName: .token) as? String, "trm-00001")
+        let bankAccount = transferMethods?.data.first { ($0.type ?? "")  == "BANK_ACCOUNT" }
+
+        XCTAssertEqual(bankAccount?.type, "BANK_ACCOUNT")
+        XCTAssertEqual(bankAccount?.token, "trm-00001")
+        XCTAssertEqual(bankAccount?.status, "ACTIVATED")
+        XCTAssertEqual(bankAccount?.createdOn, "2018-12-15T00:30:12")
+        XCTAssertEqual(bankAccount?.transferMethodCountry, "US")
+        XCTAssertEqual(bankAccount?.transferMethodCurrency, "USD")
         XCTAssertEqual(bankAccount?.getField(fieldName: .bankAccountId) as? String, "25589087")
```

Comment:
@jsaini-hw 
As `transferMethods` is of type `HyperwalletPageList<HyperwalletTransferMethod>?`, only instances of base class `HyperwalletTransferMethod` instantiated and attempts to cast down fail.
Leaving as it is for now, will appreciate for a piece of advice.

Reply:
Same as above

Reply:
Thanks, fixed